src/main/java/net/patchworkmc/patcher/util/ResourceDownloadUtil.java

```diff
@@ -0,0 +1,169 @@
+			downloadIntermediary(intermediary);
+			IMappingProvider intermediaryProvider = TinyUtils.createTinyMappingProvider(intermediary, "official", "intermediary");
+
+			List<TsrgClass<RawMapping>> classes = Tsrg.readMappings(new FileInputStream(srg.toFile()));
+			TsrgMappings tsrgMappings = new TsrgMappings(classes, intermediaryProvider);
+			this.srg = tsrgMappings;
+			this.bridged = new BridgedMappings(tsrgMappings, intermediaryProvider);
+		}
+
+		if (voldemapBridged != null) {
+			TinyWriter tinyWriter = new TinyWriter("srg", "intermediary");
+			this.bridged.load(tinyWriter);
+			Files.write(voldemapBridged, tinyWriter.toString().getBytes(StandardCharsets.UTF_8));
+		}
+
+		return bridged;
+	}
+
+	private void downloadMinecraftJars(Path client, Path server) throws IOException {
+		Path versionManifest = tempDir.resolve("mc-version-manifest.json");
+		FileUtils.copyURLToFile(new URL("https://launchermeta.mojang.com/mc/game/version_manifest.json"), versionManifest.toFile());
+
+		Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+		JsonArray versions = gson.fromJson(new String(Files.readAllBytes(versionManifest), StandardCharsets.UTF_8), JsonObject.class).get("versions").getAsJsonArray();
+
+		for (JsonElement jsonElement : versions) {
+			if (jsonElement.isJsonObject()) {
+				JsonObject object = jsonElement.getAsJsonObject();
+				String id = object.get("id").getAsJsonPrimitive().getAsString();
+
+				if (id.equals(VersionUtil.getMinecraftVersion())) {
+					String versionUrl = object.get("url").getAsJsonPrimitive().getAsString();
+					JsonObject downloads = gson.fromJson(new InputStreamReader(new URL(versionUrl).openStream()), JsonObject.class)
+							.getAsJsonObject("downloads");
+					String clientJarUrl = downloads.getAsJsonObject("client").get("url").getAsJsonPrimitive().getAsString();
+					String serverJarUrl = downloads.getAsJsonObject("server").get("url").getAsJsonPrimitive().getAsString();
+					LOGGER.trace(": downloading client jar");
+					FileUtils.copyURLToFile(new URL(clientJarUrl), client.toFile());
+					LOGGER.trace(": downloading server jar");
+					FileUtils.copyURLToFile(new URL(serverJarUrl), server.toFile());
+					break;
+				}
+			}
+		}
+	}
+
+	private void downloadSrg(Path mcp) throws IOException, URISyntaxException {
+		LOGGER.trace("      : downloading SRG");
+		Path mcpConfig = tempDir.resolve("mcp-config.zip");
+		FileUtils.copyURLToFile(new URL(FORGE_MAVEN + "/de/oceanlabs/mcp/mcp_config/" + VersionUtil.getMcpConfigVersion()
+				+ "/mcp_config-" + VersionUtil.getMcpConfigVersion() + ".zip"), mcpConfig.toFile());
+		// the jar loader opens zips just fine
+		URI inputJar = new URI("jar:" + mcpConfig.toUri());
+
+		try (FileSystem fs = FileSystems.newFileSystem(inputJar, Collections.emptyMap())) {
+			Files.copy(fs.getPath("/config/joined.tsrg"), mcp);
+		}
+	}
+
+	private void downloadIntermediary(Path intermediary) throws IOException, URISyntaxException {
```

Comment:
This seems like a copy of downloadSrg. Can these functions be deduplicated?

Reply:
It would be much less readable; you'd have to pass in a lot of extra information